File: packages/expo-video/ios/Enums/FullscreenOrientation.swift
// Copyright 2025-present 650 Industries. All rights reserved.

import ExpoModulesCore

internal enum FullscreenOrientation: String, Enumerable {
  case landscape
  case portrait
  case landscapeLeft
  case landscapeRight
  case portraitUp
  case portraitDown
  case `default`

  #if !os(tvOS)
  func toUIInterfaceOrientationMask() -> UIInterfaceOrientationMask {
    switch self {
    case .landscape:
      return .landscape
    case .portrait:
      return [.portrait, .portraitUpsideDown]
    case .landscapeLeft:
      return .landscapeLeft
    case .landscapeRight:
      return .landscapeRight
    case .portraitUp:
      return .portrait
    case .portraitDown:
      return .portraitUpsideDown
    case .default:
      return .all
    }
  }
  #endif
}

File: packages/expo-video/android/src/main/java/expo/modules/video/enums/FullscreenOrientation.kt
package expo.modules.video.enums
import expo.modules.kotlin.types.Enumerable
import android.content.pm.ActivityInfo

enum class FullscreenOrientation(val value: String) : Enumerable {
  LANDSCAPE("landscape"),
  PORTRAIT("portrait"),
  LANDSCAPE_LEFT("landscapeLeft"),
  LANDSCAPE_RIGHT("landscapeRight"),
  PORTRAIT_UP("portraitUp"),
  PORTRAIT_DOWN("portraitDown"),
  DEFAULT("default");

  fun toActivityOrientation(): Int {
    return when (this) {
      LANDSCAPE -> ActivityInfo.SCREEN_ORIENTATION_USER_LANDSCAPE
      PORTRAIT -> ActivityInfo.SCREEN_ORIENTATION_USER_PORTRAIT
      LANDSCAPE_LEFT -> ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
      LANDSCAPE_RIGHT -> ActivityInfo.SCREEN_ORIENTATION_REVERSE_LANDSCAPE
      PORTRAIT_UP -> ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
      PORTRAIT_DOWN -> ActivityInfo.SCREEN_ORIENTATION_REVERSE_PORTRAIT
      DEFAULT -> ActivityInfo.SCREEN_ORIENTATION_UNSPECIFIED
    }
  }
}
